applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1122,6 +1317,67 @@ namespace Kratos
     //************************************************************************************
     //************************************************************************************
 
+    void BaseSolidElement::CalculateAndAddKm(
```

Comment:
this function is simply hiding the following:

    Matrix tmp = prod(C,B)
    noalias(LHS) += weight*prod(trans(B),C)

if you want to leave it as a function then please implement in there the optimized Bt C B product (there are much faster algorithms for the Bt C B product which for example avoid the creation of the tmp, however if you don't do that it is definitely clearer if you leve the multiplication explicit